webcomponents/tool/src/main/frontend/packages/sakai-rubrics/src/SakaiRubric.js

```diff
@@ -262,9 +262,12 @@ export class SakaiRubric extends RubricsElement {
   }
 
   handleSaveWeights() {
-    const saveWeightsBtn = document.querySelector(`[rubric-id='${this.rubric.id}'] .save-weights`);
-    // Disable the save button
-    if (saveWeightsBtn) saveWeightsBtn.setAttribute("disabled", true);
+    // Use an event to notify SakaiRubricCriteria to disable the button
+    this.dispatchEvent(new CustomEvent("weights-saving", {
```

Comment:
I'm confused here. Is "this" referring to the criteria instance because that is the component which is firing save-weights? If true I'd just remove the weights-saving event and just assume that in the original event emitting component, SakaiRubricCriteria.